### PR TITLE
[BUGFIX] `argilla`: reusable settings objects

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR changes the behavior when working with settings. When settings are assigned to a dataset, a copy is created internally to avoid problems sharing the internal state, making the flows of duplicate datasets smoother.

For example, this is valid with this change:
```python
dataset = Dataset(
    name=<dataset_name>,
    settings=Settings(
        fields=[TextField(name="text")],
        questions=[RatingQuestion(name="question", values=[1, 2, 3, 4, 5])],
    ),
).create()

dataset.records.log([{"text": "This is a text"}])

new_dataset = Dataset(
    name=<other-dataset-name>,
    settings=dataset.settings,
).create()

new_dataset.records.log(list(dataset.records))
```

Before this change, this was not possible since the internal settings state is shared between datasets.

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by frascuchon** [comment]
Doc sections are not affected by this change. I'm not sure if an explicit entry about how to copy datasets makes sense or can be natural from how the SDK is used. cc @davidberenstein1957 @nataliaElv @dvsrepo 

https://argilla-io.github.io/argilla/dev/how_to_guides/dataset/#create-a-dataset

**Comment by nataliaElv** [comment]
> Doc sections are not affected by this change. I'm not sure if an explicit entry about how to copy datasets makes sense or can be natural from how the SDK is used. cc @davidberenstein1957 @nataliaElv @dvsrepo
> 
> https://argilla-io.github.io/argilla/dev/how_to_guides/dataset/#create-a-dataset

Yes! I've had doubts in the past about the best way to copy a dataset, so a section in that guide would be really helpful.

File: argilla/tests/integration/test_create_datasets.py [hunk 1]
```diff
@@ -92,3 +92,59 @@ def test_create_dataset_from_existing_dataset(self, client: Argilla):
...
+        dataset_name = f"test_dataset_{uuid.uuid4()}"
+        Dataset(
+            name=dataset_name,
+            settings=Settings(
+                fields=[TextField(name="text")],
+                questions=[RatingQuestion(name="question", values=[1, 2, 3, 4, 5])],
+            ),
+        ).create()
+
+        dataset = client.datasets(dataset_name)
```
Comment (davidberenstein1957) [inline argilla/tests/integration/test_create_datasets.py]:
```suggestion
        dataset = Dataset(
            name=dataset_name,
            settings=Settings(
                fields=[TextField(name="text")],
                questions=[RatingQuestion(name="question", values=[1, 2, 3, 4, 5])],
            ),
        ).create()
```

**Comment by frascuchon** [comment]
> > Doc sections are not affected by this change. I'm not sure if an explicit entry about how to copy datasets makes sense or can be natural from how the SDK is used. cc @davidberenstein1957 @nataliaElv @dvsrepo
> > https://argilla-io.github.io/argilla/dev/how_to_guides/dataset/#create-a-dataset
> 
> Yes! I've had doubts in the past about the best way to copy a dataset, so a section in that guide would be really helpful.

Done https://github.com/argilla-io/argilla/pull/5236/commits/b940c3723ced9d2a0b7d22e5b3f6efceccf7f28b

**Comment by frascuchon** [comment]
Merging this to prepare v2.0.0 release

**Pull request merged**